mongo/integration/sdam_prose_test.go

```diff
@@ -108,4 +109,37 @@ func TestSDAMProse(t *testing.T) {
 			assert.Soon(t, callback, defaultCallbackTimeout)
 		})
 	})
+
+	mt.RunOpts("client waits between failed Hellos", mtest.NewOptions().MinServerVersion("4.9"), func(mt *mtest.T) {
+		// Force hello requests to fail 5 times.
+		mt.SetFailPoint(mtest.FailPoint{
+			ConfigureFailPoint: "failCommand",
+			Mode: mtest.FailPointMode{
+				Times: 5,
+			},
+			Data: mtest.FailPointData{
+				FailCommands: []string{internal.LegacyHello, "hello"},
+				ErrorCode:    1234,
```

Comment:
Could be any value I think. `1234` was just a [placeholder](https://github.com/mongodb/specifications/commit/6c5b2acbf5a30b2c79e904532d51ba9ed8783247#diff-023626308692bf0090aadd07b8ac54a169e1d7e7c6bdc621d23c97c848081b4cR198) from the spec tests. I could remove it since we don't assert anything about the error code, but I'll probably just leave it in for consistency with other drivers.

Reply:
Ah, I didn't realize that value came directly from the prose test. Should we consider adding links to the relevant [prose test](https://github.com/mongodb/specifications/blob/88c583457c2481871ca6ab22be117a4da9d97135/source/server-discovery-and-monitoring/server-discovery-and-monitoring-tests.rst#monitors-sleep-at-least-minheartbeatfreqencyms-between-checks) or the [GODRIVER ticket](https://jira.mongodb.org/browse/GODRIVER-1824) to keep a record of what tests cover what prose tests?

Reply:
The GODRIVER ticket already links to a [DRIVERS ticket](https://jira.mongodb.org/browse/DRIVERS-1251) with the specified prose test. However, I do think our prose tests could be better labeled/numbered, so I filed GODRIVER-2274. Probably out of scope for this PR, though.